game-authorization-score/src/main/java/com/iconbet/score/authorization/Authorization.java

```diff
@@ -0,0 +1,965 @@
+				
+		//All fields should be provided       
+        
+        for (JsonValue item : array) {
+            for (Member member : item.asObject()) {
+            	 String field = member.getName();
+            	 if (!METADATA_FIELDS.contains(field)) {
+       				Context.revert("There is no "+field+" for the game");
+            	 }
+            }		
+        }
+        
+		if ( this.apply_watch_dog_method.get()) {
+			String maxPayoutStr = getValueFromItem(array,"maxPayout");
+			if (!maxPayoutStr.isBlank()) {
+				BigInteger maxPayout = new BigInteger(maxPayoutStr);
+            	if (maxPayout.compareTo(_1_ICX) == -1) {
+            		Context.revert(maxPayout.toString()+" is less than 0.1 ICX");
+            	}		
+			}else {
+				Context.revert("There is no maxPayout for the game");
+			}
+		}
+		
+		// Check if name is empty
+		String nameStr = getValueFromItem(array,"name");
+		if (nameStr.isBlank()) {
+			Context.revert("Game name cant be empty");
+		}
+		
+		// check if scoreAddress is a valid contract address
+		String scoreAddressStr = getValueFromItem(array,"name");
+		if (!scoreAddressStr.isBlank()) {
+			Address scoreAddress = Address.fromString(scoreAddressStr);
+			if (!scoreAddress.isContract()) {
+				Context.revert(scoreAddress.toString() +" is not a valid contract address");
+			}
+		}
+		
+		// Check if minbet is within defined limit of 0.1 ICX
+		String minBetStr = getValueFromItem(array,"minBet");
+		BigInteger minBet = new BigInteger(minBetStr);
+		if (minBet.compareTo(_1_ICX) == -1 ) {
+			Context.revert(minBet.toString() +" is less than 0.1 ICX");
+		}
+		
+	    // Check if proper game type is provided		
+		String gameType = getValueFromItem(array,"gameType");
+		if (!GAME_TYPE.contains(gameType)) {
+			Context.revert("Not a valid game type");
+		}
+		
+		// Check for revenue share wallet address
+		String revwallet = getValueFromItem(array,"revwallet");
+		try {
+			Address revWalletAddress = Address.fromString(revwallet);
+			if (!revWalletAddress.isContract() ) {
+				Context.revert("Not a wallet address");
+			}
+		}catch(Exception e) {
+			Context.revert("Invalid address");
+		}
+	}
+	
+	
+	@External
+	public void accumulate_daily_wagers(Address game, BigInteger wager ) {
+		/***
+        Accumulates daily wagers of the game. Updates the excess of the game.
+        Only roulette score can call this function.
+        :param game: Address of the game
+        :type game: :class:`iconservice.base.address.Address`
+        :param wager: Wager amount of the game
+        :type wager: int
+        :return:
+		***/
+		Address sender = Context.getCaller();
+		
+		if (!sender.equals(this.roulette_score.get()) ) {
+			Context.revert("Only roulette score can invoke this method.");			
+		}
+		BigInteger now = BigInteger.valueOf(Context.getBlockTimestamp());
+		BigInteger day = now.divide(U_SECONDS_DAY);
+		
+		//Question ? day.intValue() number of the day
+		BigInteger wagerValue = this.wagers.get(game)[day.intValue()];
```

Comment:
you have a lot of null pointer exception issues
you init the dictdb but has not init the BigInteger[] 
if this.wagers.get(game) returns null then accesing [day.iniValue()] will throws nullpointerex

same case for variables: 
payouts
games_excess_history

you need to verify
1) if the method is writable @External then 
```
if (this.wagers.get(game) == null ) 
      this.wagers.set(game, new BigInteger[]{ZERO,ZERO,ZERO,..., <number of enums in the structure>}) 
// use this.wagers.get(game)[X]
```

2) if the method is read only @External(readOnly=true)
```
BigInteger[] list = new BigInteger[]{ZERO,ZERO,ZERO,..., <number of enums in the structure>};
if (this.wagers.get(game) != null ) 
     list= this.wagers.get(game)

//use ... list[X]...

```       
if the array is too big, replace BigInteger[] by LinkedList<BigInteger> so you just use the necessary space

Reply:
updated accordingly.